src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -434,37 +434,107 @@ def onTabCellEdit(self, row, column):
 
     def onConstraintChange(self, row, column):
         """
-        Modify the constraint's "active" instance variable.
+        Modify the constraint when the user edits the constraint list. If the
+        user changes the constrained parameter, the constraint is erased and a
+        new one is created.
+        Checking is performed on the constrained entered by the user, showing
+        message box warning him the constraint is not valid and cancelling
+        his changes by reloading the view. View is reloaded
+        when the user is finished for consistency.
         """
         item = self.tblConstraints.item(row, column)
-        if column == 0:
-            # Update the tabs for fitting list
-            constraint = self.available_constraints[row]
-            constraint.active = (item.checkState() == QtCore.Qt.Checked)
-            param = item.data(0)[item.data(0).index(':') + 1:item.data(0).index('=')].strip()
-            model = item.data(0)[:item.data(0).index(':')].strip()
-            tab = self.available_tabs[model]
-            if isinstance(tab, FittingWidget):
-            # Update the fitting widget whenever a constraint is activated/deactivated
-                if item.checkState() == QtCore.Qt.Checked:
-                    font = QtGui.QFont()
-                    font.setItalic(True)
-                    brush = QtGui.QBrush(QtGui.QColor('blue'))
-                    tab.modifyViewOnRow(tab.getRowFromName(param), font=font, brush=brush)
-                else:
-                    tab.modifyViewOnRow(tab.getRowFromName(param))
-                return
-        # Update the constraint formula
+        # extract information from the constraint object
         constraint = self.available_constraints[row]
-        function = item.text()
-        # remove anything left of '=' to get the constraint
-        function = function[function.index('=')+1:]
-        # No check on function here - trust the user (R)
-        if function != constraint.func:
-            # This becomes rather difficult to validate now.
-            # Turn off validation for Edit Constraint
-            constraint.func = function
-            constraint.validate = False
+        model = constraint.value_ex[:constraint.value_ex.index(".")]
+        param = constraint.param
+        function = constraint.func
+        tab = self.available_tabs[model]
+        # Extract information from the text in the table
+        constraint_text = item.data(0)
+        # Basic sanity check of the string
+        # First check if we have an acceptable sign
```

Comment:
This is a minefield. I can't imagine assuring a freeform constraint validator working without a massive dictionary of possible options...
One solution would be to replace `model:parameter` with a number and attempting to evaluate such an expression? I think I've done it somewhere else for constraints.